src/System.Device.Gpio/System/Device/Pwm/Drivers/Windows10PwmDriverChip.Windows.cs

```diff
@@ -22,7 +23,18 @@ public Windows10PwmDriverChip(int chipIndex)
             DeviceInformationCollection deviceInformationCollection = DeviceInformation.FindAllAsync(deviceSelector).WaitForCompletion();
             if (deviceInformationCollection.Count == 0)
             {
-                throw new ArgumentException($"No PWM device exists for PWM chip at index {chipIndex}", $"{nameof(chipIndex)}");
+                // On Hummingboard, fallback to check for generic PWM controller (friendlyname is not currently used)
+                var deviceInfo = new EasClientDeviceInformation();

```

Comment:
I wonder if we should do this check on the PWMController instead, and just pass it into the Chip? That way we would only make the check once per controller, instead of one per chip?

Reply:
I'm not sure I follow. Which function are you suggesting to make the check?

Reply:
Sorry it wasn't super clean. What I meant, is that the only call-site for this method is here: https://github.com/dotnet/iot/blob/e8bcd119cb2b102bf17d2b66158aa752d3952817/src/System.Device.Gpio/System/Device/Pwm/Drivers/Windows10PwmDriver.Windows.cs#L17

If we have the check done on the main driver, then we don't have to perform the check every time we open a new DriverChip, and we can pass that result down to each DriverChip so that in case we are running on a Hummingboard we default to the Chip0. Does that make sense?

Reply:
Ok, so perform the check for Hummingboard in the OpenChannel() function, and pass a Boolean variable to Windows10PwmDriverChip() to indicate to use the default channel. Is that what you meant?

Reply:
right, and of course you should only make the check once, so you should cache whatever the result was, that way you can open how ever many channels you want but you will only be checking whether or not to use the default only once.

Reply:
updated to check the board type only once in the constructor